esp32-mock-generator/src/main.cpp

```
@@ -0,0 +1,319 @@
+// ============================================================================
+// Generate and Publish Measurement
+// ============================================================================
+void generateAndPublishMeasurement() {
+  measurementCount++;
+
+  // -------------------------------------------------------------------------
+  // 1. Generate Base Electrical Parameters
+  // -------------------------------------------------------------------------
+
+  // Voltage: normally around 230V ±5V, with occasional sags/swells
+  float voltage = NOMINAL_VOLTAGE + randomFloat(-5.0, 5.0);
```

Clang (14.0.6)

[warning] 5.0 is a magic number; consider replacing it with a named constant

(cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers)